src/sentry/similarity/index.py

```diff
@@ -273,8 +243,30 @@ def import_(self, scope, items, timestamp=None):
         for idx, key, data in items:
             arguments.extend([idx, key, data])
 
-        return index(
-            self._get_connection(scope),
-            [],
-            arguments,
-        )
+        return self.__index(scope, arguments)
+
+
+class NoopIndex(object):
```

Comment:
We should more strictly define these return types for these methods, I don't think the ones here line up 1:1.

Reply:
I updated these here 311d030431adc4a22ea2242b2fbc86b649620dab

However, I'm still not positive that they accurately reflect the return types from. For example, scan isn't a generator, but just returns a list. What do you think this should look like?

Reply:
I'd need to double check what some of the Lua translations are (e.g. is `nil` translated to `False` or `None` — you may have already done this for `merge` and `delete`, but I haven't verified the result.)

For `scan` you can force it to return an empty generator like this: https://github.com/getsentry/sentry/blob/311d030431adc4a22ea2242b2fbc86b649620dab/src/sentry/digests/backends/dummy.py#L19-L21

Reply:
I generally think `abc.ABCMeta` is borderline useless (since it can't enforce anything other than that a method with the same name exists, no parameter or return type validation), but this might be a case where it at least provides a little utility: https://docs.python.org/2/library/abc.html#abc.ABCMeta

Reply:
484d7f234fe2c11704f283d7ac2f0c85f13333b1 for the iterator
83fd56252afcfdc63e83129079036ba8ec5ba418 / 7a95aee04c9bc7545e6888e11ac200dd47309f09 for missing `import_`